From 24f65fcb0101a49ed618b9dbb96fa2d370a10d14 Mon Sep 17 00:00:00 2001
From: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
Date: Fri, 28 Mar 2025 10:02:42 +0700
Subject: [PATCH] rzt2: linux-renesas: Add support changing ports to 
 MII<->RGMII for RZN2H board.

This patch support changing ports to MII<->RGMII via config
MACHINE_FEATURES_append on local.conf for RZN2H board.

Signed-off-by: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
---
 ...n2h-dev-Config-100-mbps-Ethernet-for.patch | 91 ++++++++++++++++++
 .../u-boot/u-boot_2021.10.bbappend            |  1 +
 ...as-r9a09g087m44-dev-Config-ETH0-ETH1.patch | 95 +++++++++++++++++++
 ...as-r9a09g087m44-dev-Config-ETH2-ETH3.patch | 78 +++++++++++++++
 .../linux/linux-renesas_5.10.bbappend         |  2 +
 5 files changed, 267 insertions(+)
 create mode 100644 meta-rzt2h/recipes-bsp/u-boot/u-boot/0002-board-renesas-rzn2h-dev-Config-100-mbps-Ethernet-for.patch
 create mode 100644 meta-rzt2h/recipes-kernel/linux/linux-renesas/0003-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH0-ETH1.patch
 create mode 100644 meta-rzt2h/recipes-kernel/linux/linux-renesas/0004-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH2-ETH3.patch

diff --git a/meta-rzt2h/recipes-bsp/u-boot/u-boot/0002-board-renesas-rzn2h-dev-Config-100-mbps-Ethernet-for.patch b/meta-rzt2h/recipes-bsp/u-boot/u-boot/0002-board-renesas-rzn2h-dev-Config-100-mbps-Ethernet-for.patch
new file mode 100644
index 00000000..b878b552
--- /dev/null
+++ b/meta-rzt2h/recipes-bsp/u-boot/u-boot/0002-board-renesas-rzn2h-dev-Config-100-mbps-Ethernet-for.patch
@@ -0,0 +1,91 @@
+From 9073c714f98f96eb7fce87fda989e616f70231c7 Mon Sep 17 00:00:00 2001
+From: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+Date: Thu, 27 Mar 2025 14:01:21 +0700
+Subject: [PATCH] board: renesas: rzn2h-dev: Config 100 mbps Ethernet for ETH2,
+ ETH3 ports on RZ/N2H boards
+
+This patch config 100 mbps Ethernet for ETH2, ETH3 ports
+on RZ/N2H boards.
+
+Signed-off-by: Long Luu <long.luu.ur@renesas.com>
+Signed-off-by: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+---
+ arch/arm/dts/rzn2h-dev.dts          | 10 ++++++----
+ board/renesas/rzn2h-dev/rzn2h-dev.c | 12 ++++++------
+ 2 files changed, 12 insertions(+), 10 deletions(-)
+
+diff --git a/arch/arm/dts/rzn2h-dev.dts b/arch/arm/dts/rzn2h-dev.dts
+index a5b5019..1edc183 100644
+--- a/arch/arm/dts/rzn2h-dev.dts
++++ b/arch/arm/dts/rzn2h-dev.dts
+@@ -80,7 +80,8 @@
+ 
+ &gmac1 {
+ 	phy-handle = <&phy3>;
+-	phy-mode = "rgmii-id";
++	phy-mode = "mii";
++	max-speed = <100>;
+ 	status = "okay";
+ 
+ 	mdio {
+@@ -88,7 +89,7 @@
+ 		#size-cells = <0>;
+ 		phy3: ethernet-phy@3 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "rgmii-id";
++			phy-mode = "mii";
+ 			reg = <3>;
+ 		};
+ 	};
+@@ -96,7 +97,8 @@
+ 
+ &gmac2 {
+ 	phy-handle = <&phy2>;
+-	phy-mode = "rgmii-id";
++	phy-mode = "mii";
++	max-speed = <100>;
+ 	status = "okay";
+ 
+ 	mdio {
+@@ -104,7 +106,7 @@
+ 		#size-cells = <0>;
+ 		phy2: ethernet-phy@2 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "rgmii-id";
++			phy-mode = "mii";
+ 			reg = <2>;
+ 		};
+ 	};
+diff --git a/board/renesas/rzn2h-dev/rzn2h-dev.c b/board/renesas/rzn2h-dev/rzn2h-dev.c
+index fae8455..0c291ea 100644
+--- a/board/renesas/rzn2h-dev/rzn2h-dev.c
++++ b/board/renesas/rzn2h-dev/rzn2h-dev.c
+@@ -334,19 +334,19 @@ int board_init(void)
+ 	/* ETHSS: Mode Control 0x6, GMAC1 on port ETH3, GMAC2 on port ETH2 */
+ 	ethss_init_hw(0x6);
+ 
+-	/* ETHSS: GMAC1 RGMII_ID mode on port ETH3 */
+-	ret = ethss_config(3, PHY_INTERFACE_MODE_RGMII_ID);
++	/* ETHSS: GMAC1 MII mode on port ETH3 */
++	ret = ethss_config(3, PHY_INTERFACE_MODE_MII);
+ 	if (ret < 0)
+ 		return ret;
+ 	/* Set up speed for Converters for GMAC1 */
+-	ethss_link_up(3, PHY_INTERFACE_MODE_RGMII_ID, SPEED_1000, DUPLEX_FULL);
++	ethss_link_up(3, PHY_INTERFACE_MODE_MII, SPEED_100, DUPLEX_FULL);
+ 
+-	/* ETHSS: GMAC2 RGMII_ID mode on port ETH2 */
+-	ret = ethss_config(2, PHY_INTERFACE_MODE_RGMII_ID);
++	/* ETHSS: GMAC2 MII mode on port ETH2 */
++	ret = ethss_config(2, PHY_INTERFACE_MODE_MII);
+ 	if (ret < 0)
+ 		return ret;
+ 	/* Set up speed for Converters for GMAC2 */
+-	ethss_link_up(2, PHY_INTERFACE_MODE_RGMII_ID, SPEED_1000, DUPLEX_FULL);
++	ethss_link_up(2, PHY_INTERFACE_MODE_MII, SPEED_100, DUPLEX_FULL);
+ 
+ 	board_usb_init();
+ 
+-- 
+2.7.4
+
diff --git a/meta-rzt2h/recipes-bsp/u-boot/u-boot_2021.10.bbappend b/meta-rzt2h/recipes-bsp/u-boot/u-boot_2021.10.bbappend
index 7c5a9329..bc8b4869 100644
--- a/meta-rzt2h/recipes-bsp/u-boot/u-boot_2021.10.bbappend
+++ b/meta-rzt2h/recipes-bsp/u-boot/u-boot_2021.10.bbappend
@@ -5,6 +5,7 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 SRC_URI_append_rzt2h = " \
 	${@bb.utils.contains('MACHINE_FEATURES', 'eth23_mii', ' \
 	file://0001-board-renesas-rzt2h-dev-Config-100-mbps-Ethernet-for.patch \
+	file://0002-board-renesas-rzn2h-dev-Config-100-mbps-Ethernet-for.patch \
 	', '', d)} \
 "
 
diff --git a/meta-rzt2h/recipes-kernel/linux/linux-renesas/0003-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH0-ETH1.patch b/meta-rzt2h/recipes-kernel/linux/linux-renesas/0003-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH0-ETH1.patch
new file mode 100644
index 00000000..c7b5b665
--- /dev/null
+++ b/meta-rzt2h/recipes-kernel/linux/linux-renesas/0003-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH0-ETH1.patch
@@ -0,0 +1,95 @@
+From ca56794da2e7106297e515acdb26afc7410638a7 Mon Sep 17 00:00:00 2001
+From: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+Date: Thu, 27 Mar 2025 13:31:20 +0700
+Subject: [PATCH 1/2] arm64: dts: renesas: r9a09g087m44-dev: Config ETH0, ETH1
+ ports to RGMII mode
+
+This patch change ETH0, ETH1 ports to RGMII mode.
+
+Signed-off-by: Long Luu <long.luu.ur@renesas.com>
+Signed-off-by: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+---
+ arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts | 20 ++++++++------------
+ 1 file changed, 8 insertions(+), 12 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts b/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
+index fa6bbef..0ed6614 100644
+--- a/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
++++ b/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
+@@ -948,8 +948,7 @@
+ 
+ 		phy0: ethernet-phy@0 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "mii";
+-			max-speed = <100>;
++			phy-mode = "rgmii-id";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -958,8 +957,7 @@
+ 
+ 		phy1: ethernet-phy@1 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "mii";
+-			max-speed = <100>;
++			phy-mode = "rgmii-id";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -970,14 +968,14 @@
+ 
+ &switch_port0 {
+ 	label = "lan0";
+-	phy-mode = "mii";
++	phy-mode = "rgmii-id";
+ 	phy-handle = <&phy0>;
+ 	status = "okay";
+ };
+ 
+ &switch_port1 {
+ 	label = "lan1";
+-	phy-mode = "mii";
++	phy-mode = "rgmii-id";
+ 	phy-handle = <&phy1>;
+ 	status = "okay";
+ };
+@@ -1001,8 +999,7 @@
+ 
+ 		phy0: ethernet-phy@0 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "mii";
+-			max-speed = <100>;
++			phy-mode = "rgmii-id";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -1011,8 +1008,7 @@
+ 
+ 		phy1: ethernet-phy@1 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "mii";
+-			max-speed = <100>;
++			phy-mode = "rgmii-id";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -1043,14 +1039,14 @@
+ 
+ &esc_port0 {
+ 	label = "esc0";
+-	phy-mode = "mii";
++	phy-mode = "rgmii-id";
+ 	phy-handle = <&phy0>;
+ 	status = "okay";
+ };
+ 
+ &esc_port1 {
+ 	label = "esc1";
+-	phy-mode = "mii";
++	phy-mode = "rgmii-id";
+ 	phy-handle = <&phy1>;
+ 	status = "okay";
+ };
+-- 
+2.7.4
+
diff --git a/meta-rzt2h/recipes-kernel/linux/linux-renesas/0004-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH2-ETH3.patch b/meta-rzt2h/recipes-kernel/linux/linux-renesas/0004-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH2-ETH3.patch
new file mode 100644
index 00000000..c675852b
--- /dev/null
+++ b/meta-rzt2h/recipes-kernel/linux/linux-renesas/0004-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH2-ETH3.patch
@@ -0,0 +1,78 @@
+From bff871afc87b43a65aab6b412ae84dc7f1d498ec Mon Sep 17 00:00:00 2001
+From: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+Date: Thu, 27 Mar 2025 13:37:17 +0700
+Subject: [PATCH 2/2] arm64: dts: renesas: r9a09g087m44-dev: Config ETH2, ETH3
+ ports to MII mode
+
+This patch change ETH2, ETH3 ports to MII mode.
+
+Signed-off-by: Long Luu <long.luu.ur@renesas.com>
+Signed-off-by: Vo Dai Mau <mau.vo-dai@banvien.com.vn>
+---
+ arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts | 15 +++++++++------
+ 1 file changed, 9 insertions(+), 6 deletions(-)
+
+diff --git a/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts b/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
+index 0ed6614..e292385 100644
+--- a/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
++++ b/arch/arm64/boot/dts/renesas/r9a09g087m44-dev.dts
+@@ -858,7 +858,8 @@
+ #if (!SEL_PCIE && !SEL_SPI2 && !SEL_ADC)
+ &gmac1 {
+ 	phy-handle = <&phy3>;
+-	phy-mode = "rgmii-id";
++	phy-mode = "mii";
++	max-speed = <100>;
+ 	pcs-handle = <&mii_conv3>;
+ 	status = "okay";
+ 	phy-reset-gpios = <&pinctrl RZT2H_GPIO(3, 2) GPIO_ACTIVE_HIGH>;
+@@ -875,7 +876,7 @@
+ 		#size-cells = <0>;
+ 		phy3: ethernet-phy@3 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "rgmii-id";
++			phy-mode = "mii";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -888,7 +889,8 @@
+ #if (!SEL_ESC_ETHSW_GMAC2)
+ &gmac2 {
+ 	phy-handle = <&phy2>;
+-	phy-mode = "rgmii-id";
++	phy-mode = "mii";
++	max-speed = <100>;
+ 	pcs-handle = <&mii_conv2>;
+ 	status = "okay";
+ 	phy-reset-gpios = <&pinctrl RZT2H_GPIO(3, 1) GPIO_ACTIVE_HIGH>;
+@@ -905,7 +907,7 @@
+ 		#size-cells = <0>;
+ 		phy2: ethernet-phy@2 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "rgmii-id";
++			phy-mode = "mii";
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -1017,7 +1019,8 @@
+ 
+ 		phy2: ethernet-phy@2 {
+ 			compatible = "ethernet-phy-id0007.0772";
+-			phy-mode = "rgmii-id";
++			phy-mode = "mii";
++			max-speed = <100>;
+ 			vsc8531,led-0-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-mode = <VSC8531_LINK_ACTIVITY>;
+ 			vsc8531,led-1-combine-disable;
+@@ -1053,7 +1056,7 @@
+ 
+ &esc_port2 {
+ 	label = "esc2";
+-	phy-mode = "rgmii-id";
++	phy-mode = "mii";
+ 	phy-handle = <&phy2>;
+ 	status = "okay";
+ };
+-- 
+2.7.4
+
diff --git a/meta-rzt2h/recipes-kernel/linux/linux-renesas_5.10.bbappend b/meta-rzt2h/recipes-kernel/linux/linux-renesas_5.10.bbappend
index 17b577d2..697ec6d6 100644
--- a/meta-rzt2h/recipes-kernel/linux/linux-renesas_5.10.bbappend
+++ b/meta-rzt2h/recipes-kernel/linux/linux-renesas_5.10.bbappend
@@ -5,9 +5,11 @@ FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}:"
 SRC_URI_append_rzt2h = " \
 	${@bb.utils.contains('MACHINE_FEATURES', 'eth01_rgmii', ' \
 		file://0001-arm64-dts-renesas-r9a09g077m44-dev-Config-ETH0-ETH1-.patch \
+		file://0003-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH0-ETH1.patch \
 	', '', d)} \
 	${@bb.utils.contains('MACHINE_FEATURES', 'eth23_mii', ' \
 		file://0002-arm64-dts-renesas-r9a09g077m44-dev-Config-ETH2-ETH3-.patch \
+		file://0004-arm64-dts-renesas-r9a09g087m44-dev-Config-ETH2-ETH3.patch \
 	', '', d)} \
 "
 
-- 
2.25.1

